packages/mbed-greentea/test/mbed_gt_cli.py

```diff
@@ -64,7 +64,7 @@ def test_get_greentea_version(self):
 
         self.assertIs(type(version), str)
         extra_info_pos = version.find('dev')
-        version = version[:extra_info_pos-1]
+        version = version[:extra_info_pos-1] if extra_info_pos > 0 else version
```

Comment:
Why can't we always use the `setuptools-scm` provided version? I don't see why we need `extra_info_pos`.

Reply:
@Patater `setuptools-scm` returns version as `version.dev*` when the tree is dirty or the commit doesn't contain a tag. These tests are run for every PR and not every PR might have a tag associated with it. Therefore we need to filter out extra information appended to version by `setuptools-scm`.

Reply:
Are there always three numbers present in the front of the version given by setuptools-scm?

Reply:
yeah, the Default versioning scheme is explained here https://pypi.org/project/setuptools-scm/

Reply:
So, we don't need to check for `dev` at all then, do we? The numbers are always present in the front and we can use those for a,b,c, no matter what comes later.

Reply:
@Patater sure, but test fails if `split()` returns more than 3 digits. That's why we only extract version information if `setuptools_scm` adds `dev*` to the version.

Reply:
That's a problem with the way the test is written.

Try this, to ignore the later values:
```
- a,b,c = version.split('.')
+ a,b,c,_ = version.split('.')
```